aea/crypto/cosmos.py

```diff
@@ -646,14 +646,15 @@ def try_execute_wasm_transaction(
                 command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT
             ).communicate()
 
-        return stdout.decode("ascii")
+        tx_digest_json = json.loads(stdout.decode("ascii"))
+        return tx_digest_json["txhash"]
 
     @staticmethod
     @try_decorator(
         "Encountered exception when trying to execute wasm query: {}",
         logger_method=_default_logger.warning,
     )
-    def try_execute_wasm_query(
+    def _try_execute_wasm_query(
```

Comment:
This needs to be wrapped in a properly named external method. Also, what is this versus get state?

Reply:
Was there an answer to this question? It also looks similar to `get_state` to me.